src/main/java/com/cinema/security/CustomUserDetailsService.java

```diff
@@ -23,11 +22,11 @@ public CustomUserDetailsService(UserService userService) {
     @Override
     public UserDetails loadUserByUsername(String name) throws UsernameNotFoundException {
         User user = userService.findByEmail(name)
-                .orElseThrow(() -> new DataProcessingException("User don't exist"));
+                .orElseThrow(() -> new UsernameNotFoundException("User don't exist"));
         org.springframework.security.core.userdetails.User.UserBuilder userBuilder =
                 org.springframework.security.core.userdetails.User.withUsername(user.getEmail());
         List<GrantedAuthority> authoritySet = user.getRoles().stream()
-                .map(r -> new SimpleGrantedAuthority(r.getName().name()))
+                .map(r -> new SimpleGrantedAuthority("ROLE_" + r.getName().name()))
```

Comment:
do we need ROLE_ now ?

Reply:
SimpleGrantedAuthority needs roles with prefix, or app will not work correctly